File: Wayne/WayneCompiler/src/Dfir_representation/ILoopIndex.java
package Dfir_representation;

import java.util.Vector;


public class ILoopIndex extends IBorderNode{
	   
	   public ILoopIndex(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   JNodeType = "ILoopIndex";
	   }
	  
	   public void print(){
		   System.out.println("ILoopIndex begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
		   for(int i=0; i<inputTerminals.size(); i++){
				   inputTerminals.get(i).print();
		   }
		  for(int i=0; i<outputTerminals.size(); i++){
				   outputTerminals.get(i).print();
		   }   
		  System.out.println("ILoopIndex end\n");
	   }
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/ILoopIndex.cs
using System;
using System.Collections.Generic;


namespace DFIR_Compiler
{
    public class ILoopIndex:IBorderNode
    {
        public ILoopIndex(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            INodeType = "ILoopIndex";
        }

        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("ILoopIndex:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
            for (int i = 0; i < inputTerminals.Count; i++)
            {
                inputTerminals[i].Print();
            }
            for (int i = 0; i < outputTerminals.Count; i++)
            {
                outputTerminals[i].Print();
            }
        }
    }
}
